### ISSUE TITLE
Documentation !!

### ISSUE THREAD

**Issue by robross0606** [deepsidhu1313/SyntaxTextAreaFX#3]
Looking at this project structure, what is the point of all the JSON files?  As far as I can tell, they aren't used at all in the code base.  The only files that are syntax highlighted are the ones that have definitions defined in in.co.s13.langs.

**Comment by robross0606** [comment]
There are also two sets of JSON files in the source (under "jsons" and "json2").  What is the purpose of each?

**Comment by deepsidhu1313** [comment]
JSON directory in the root folder is the one with all the rules for different languages. And as you can see  [json2](https://github.com/deepsidhu1313/SyntaxTextAreaFX/blob/master/src/in/co/s13/res/json2/R.json)
directory is the derived from first one, latter contains only keywords to highlight . These directories and files are for development purpose only. These will be removed after addition of all rules in their corresponding Java files under  in.co.s13.langs .

**Comment by robross0606** [comment]
I thought the approach was to directly use the JSON files as the syntax 
definition.  You're manually created Java files associated with each 
JSON file?

------ Original Message ------

**Comment by deepsidhu1313** [comment]
Initially, i thought that will be good idea. But reading data from files will have (I/O) overhead associated with it. So for basic keywords we can use hard coded rules.

**Comment by robross0606** [comment]
Do you plan on auto-generating the Java code?

------ Original Message ------

**Comment by deepsidhu1313** [comment]
if you mean suggestions by Auto-generating, then Yes i am working on that. As richtextfx are working on their next  release and i want my code to build on stable code. So i am waiting for the next stable release of RichTextFX.

**Comment by robross0606** [comment]
I mean the java classes under in.co.s13.lang.  Are those going to be 
manually created to auto-generated?

------ Original Message ------

**Comment by deepsidhu1313** [comment]
:smile:  Can you provide more detail to your question ?? As currently it is very vague or i am unable to interpret it.

**Comment by robross0606** [comment]
I mean are you intending to hand-write the .java files in the `in.co.s13.lang` package.  Or are you intending to use the JSON files to run through a code generator and produce the JAVA files to handle the various types of `Language` interface implementations.

**Comment by deepsidhu1313** [comment]
Yes i can do the code generation for that. But i rely on myself, to avoid bugs. As basic skeleton is ready one can just read files from JSON and JSON2 directory to create corresponding `.java` files under `in.co.s13.syntaxtextareafx.langs` and modify related css files in [in.co.s13.syntaxtextareafx.res.css.default](https://github.com/deepsidhu1313/SyntaxTextAreaFX/tree/master/src/in/co/s13/syntaxtextareafx/res/css/default). I will try to post a developer guide soon.

**Comment by deepsidhu1313** [comment]
Updated Wiki for syntax Highlighting support!!

For code suggestions we will have to wait for RichTextFX next stable release.
